server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -347,6 +348,11 @@ public SortFieldAndFormat build(SearchExecutionContext context) throws IOExcepti
         }
 
         MappedFieldType fieldType = context.getFieldType(fieldName);
+
+        if (fieldType instanceof RangeFieldMapper.RangeFieldType) {
+            throw new IllegalArgumentException("Sorting by range field [" + fieldName + "] is not supported");
+        }
```

Comment:
how do we do the same for other non sortable field types? Are there ways to avoid an instance of check perhaps?